src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -24,7 +24,8 @@ public sealed class NpgsqlBinaryExporter : ICancelable
 
     NpgsqlConnector _connector;
     NpgsqlReadBuffer _buf;
-    bool _isConsumed, _isDisposed;
+    // We consider COPY operation as consumed until Init successfully completes
+    bool _isDisposed, _isConsumed = true;
```

Comment:
nit: maybe introduce an ExporterState enum rather than multiple state variables? This would allow starting in an Uninitialized state and move to Initialized/Active/whatever once Init is done, rather than reusing "unconsumed" to mean "uninitialized" as this PR does (slightly confusing).